protocol/consistencychecks_test.go

```diff
@@ -46,6 +46,20 @@ func TestVerifyWithError(t *testing.T) {
 	}
 }
 
+func TestMalformedClientMessage(t *testing.T) {
+	d, pk := NewTestDirectory(t, true)
+	cc := NewCC(d.LatestSTR(), true, pk)
+
+	request := &RegistrationRequest{
+		Username: "", // invalid username
+		Key:      key,
+	}
+	res, _ := d.Register(request)
+	if err := cc.HandleResponse(RegistrationType, res, "", key); err != ErrMalformedClientMessage {
+		t.Error("Unexpected verification result")
+	}
+}
```

Comment:
We could have more tests like this. What is happens for instance if either the name or the key exceed a certain length (and what is expected to happen?)